### PR TITLE
Make level-7.

### PR DESCRIPTION
**Pull request by creator-max**
Added a stack for the garage.

### CODE REVIEW

File: AutoPark/Garage/MyStack.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        public T Pop()
+        {
+            if(Count == 0)
+            {
+                throw new InvalidOperationException();
+            }
+            var popItem = _stack[_headIndex - 1];
+            var newStack = new T[_stack.Length - 1];
+            Array.Copy(_stack, newStack, --_headIndex);
```
Comment (PushMeNow) [inline AutoPark/Garage/MyStack.cs:61]: Actually I don't see the reason to recreate array

File: AutoPark/Garage/MyStack.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        public MyStack(int size)
+        {
+            if(size <= 0)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+            _stack = new T[size];
+        }
+
+        public int Count => _headIndex;
```
Comment (PushMeNow) [inline AutoPark/Garage/MyStack.cs:40]: Count != headIndex

**Review by PushMeNow — CHANGES_REQUESTED** [review]

**Review by kuzmiich — APPROVED** [review]
good job. Could replace `==` to `is`

File: AutoPark/Garage/MyStack.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        private T[] _stack;
+
+        public MyStack()
+        {
+            _stack = new T[DefaultSize];
+        }
+
+        public MyStack(IEnumerable<T> collection)
+        {
+            if (collection == null)
```
[the remark this answers is not in the document]
Comment (creator-max) [reply AutoPark/Garage/MyStack.cs]: Replaced

File: AutoPark/Garage/MyStack.cs (hunk 1 above)
Comment (creator-max) [reply AutoPark/Garage/MyStack.cs]: I agree. Thanks, fixed.

File: AutoPark/Garage/MyStack.cs (hunk 2 above)
Comment (creator-max) [reply AutoPark/Garage/MyStack.cs]: Refused to use _headIndex.

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**